File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/GitHubRepository.java
package com.cpy3f2.Gixor.Domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * @author : simon
 * @description : GitHub仓库信息实体类
 * @since : 2024-10-30 19:43
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class GitHubRepository {
    @JsonProperty("id")
    private Long githubId;
    private String name;
    @JsonProperty("full_name")
    private String fullName;
    private Owner owner;
    private License license;
    private String description;
    @JsonProperty("open_issues")
    private Integer issues;
    private String language;
    @JsonProperty("stargazers_count")
    private Integer stargazersCount;
    @JsonProperty("forks_count")
    private Integer forksCount;
    private String visibility;
    private Boolean fork;
    @JsonProperty("html_url")
    private String htmlUrl;
    private List<String> topics;
    @JsonProperty("created_at")
    private LocalDateTime createdAt;
    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
    @JsonProperty("pushed_at")
    private LocalDateTime pushedAt;
    private Integer size;
    @JsonProperty("default_branch")
    private String defaultBranch;

    /**
     * @author : simon
     * @description : GitHub仓库所有者信息
     * @since : 2024-10-30 19:43
     */
    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Owner {
        @JsonProperty("id")
        private Long githubId;
        @JsonProperty("login")
        private String name;
        @JsonProperty("avatar_url")
        private String avatarUrl;
        @JsonProperty("html_url")
        private String htmlUrl;
        private String type;
    }

    /**
     * 仓库的许可协议信息
     * @author simon
     * @since 2024/10/31     */
    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class License {
        private String name;
        @JsonProperty("spdx_id")
        private String spdId;
        private String key;
        private String url;
    }
}

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/GitHubRepository.kt
package com.cpy3f2.gixor_mobile.model.entity

import com.google.gson.annotations.SerializedName

data class GitHubRepository(
    @SerializedName("id")
    val githubId: Long? = null,
    @SerializedName("name")
    val name: String? = null,
    @SerializedName("full_name")
    val fullName: String = "",
    @SerializedName("description")
    val description: String? = null,
    @SerializedName("language")
    val language: String? = null,
    @SerializedName("stargazers_count")
    val stargazersCount: Int? = null,
    @SerializedName("forks_count")
    val forksCount: Int? = null,
    @SerializedName("open_issues")
    val openIssues: Int? = null,
    @SerializedName("watchers_count")
    val watchersCount:Int? = null,
    val owner: Owner? = null,
    val license: License? = null,
    val visibility: String? = null,
    val fork: Boolean? = null,
    val htmlUrl: String? = null,
    val topics: List<String>? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val pushedAt: String? = null,
    val size: Int? = null,
    val defaultBranch: String? = null
) {

    data class Owner(
        val githubId: Long? = null,
        val name: String? = null,
        val avatarUrl: String? = null,
        val htmlUrl: String? = null,
        val type: String? = null
    )

    data class License(
        val name: String? = null,
        val spdId: String? = null,
        val key: String? = null,
        val url: String? = null
    )
}